docs/compilers/Co-located core types.md

```diff
@@ -15,7 +15,7 @@ Scenarios
 
 ## Design
 
-Compilers may assume that there is exactly one assembly representing the "core assembly". This core assembly is either referenced or is currently being compiled.
+Compilers may assume that there is exactly one assembly representing the "core assembly". This core assembly is either referenced or is currently being compiled. The unique core assembly may be identified by the compiler as the first referenced assembly that does not reference any other assembly, or the assembly being compiled if there are no referenced assemblies.
 
```

Comment:
Don't we check that we have a single cor lib (as opposed to picking the first one)?

Perhaps we could be more rigorous here? How about we define what core assembly means and what the requirements are on it? Such definition should imo say 

> *core assembly* is an assembly that defines System.Object
> core assembly is required to have no references